### PR TITLE
WT-12927 Avoid checkpoint cleanup processing empty or newly created files

### PR DESCRIPTION
**Pull request by kommiharibabu**
Accessing the empty or newly created files by checkpoint cleanup leads to marking them dirty unnecessarily and causing problems to open the bulk cursor when these are dirtied tables are processed by the checkpoint thread.

### CODE REVIEW

File: src/btree/bt_sync_obsolete.c [hunk 1]
```diff
@@ -477,8 +479,16 @@ __checkpoint_cleanup_eligibility(WT_SESSION_IMPL *session, const char *uri, cons
         if (ret == 0)
             newest_stop_durable_ts = WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)value.val);
         WT_RET_NOTFOUND_OK(ret);
+        ret = __wt_config_subgets(session, &cval, "addr", &value);
+        if (ret == 0)
+            addr_size = value.len;
```
Comment (etienneptl) [inline src/btree/bt_sync_obsolete.c:484]: Do we need to keep looping? Should we `break`?

File: src/btree/bt_sync_obsolete.c [hunk 2]
```diff
@@ -477,8 +479,16 @@ __checkpoint_cleanup_eligibility(WT_SESSION_IMPL *session, const char *uri, cons
...
+        WT_RET_NOTFOUND_OK(ret);
     }
 
+    /* Ignore the tables that are empty or newly created. */
+    if (addr_size == 0)
```
Comment (etienneptl) [inline src/btree/bt_sync_obsolete.c:489]: Can we add this condition to the one below and explain why we need it to be greater than 0?

Comment (kommiharibabu) [reply src/btree/bt_sync_obsolete.c]: Moved it into the below check code.

File: src/btree/bt_sync_obsolete.c (hunk 1 above)
Comment (kommiharibabu) [reply src/btree/bt_sync_obsolete.c:484]: No, we need to check all the checkpoints that existed for this table before making a decision.

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/btree/bt_sync_obsolete.c (hunk 1 above)
Comment (etienneptl) [reply src/btree/bt_sync_obsolete.c:484]: I see, thank you. Can we change the `for` loop to:
```
        while ((ret = __wt_config_next(&ckptconf, &key, &cval)) == 0) {
...
        }
WT_RET_NOTFOUND_OK(ret);
```
See an example [here](https://github.com/wiredtiger/wiredtiger/blob/5a8e337c9a79390d6be8643c7c1c2cb726ad41da/src/config/config_api.c#L222-L265).
I am suggesting this as we are currently not catching any potential error from `__wt_config_next`.

**Review by etienneptl — APPROVED** [review]

File: src/btree/bt_sync_obsolete.c (hunk 1 above)
Comment (kommiharibabu) [reply src/btree/bt_sync_obsolete.c:484]: Fixed.

**Pull request merged**